qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1738,12 +1739,19 @@ def nodes_on_wire(self, wire, only_ops=False):
             except rx.NoSuitableNeighbors:
                 pass
 
-    def count_ops(self):
+    def count_ops(self, expand_control_flow=False):
         """Count the occurrences of operation names.
 
         Returns a dictionary of counts keyed on the operation name.
         """
-        return self._op_names.copy()
+        op_count = self._op_names.copy()
+        if expand_control_flow:
+            for cfop in self.op_nodes(op=ControlFlowOp):
+                if cfop.name in op_count:
+                    del op_count[cfop.name] 
+                for block in cfop.op.blocks:
+                    op_count = Counter(op_count) + Counter(block.count_ops())
+        return op_count
```

Comment:
I did this (along with a few of the other DAGCircuit methods) in #8763 as well, so we probably only need one.  In #8763 I actually made the default `True`, so this pass would automatically update itself, though the test would be good to keep.